packages/@react-spectrum/listbox/docs/ListBox.mdx

```diff
@@ -28,35 +33,304 @@ category: Collections
+    {id: 7, name: 'Snake'},
+    {id: 8, name: 'Turtle'},
+    {id: 9, name: 'Wombat'}
+  ];
+
+  return (
+    <>
+      <ListBox
+        selectionMode="single"
+        aria-label="Pick an animal"
+        items={options}
+        onSelectionChange={selected => setAnimalId(selected)}
+        width="size-2400">
+        {item => <Item>{item.name}</Item>}
+      </ListBox>
+      <p>Your favorite animal has id: {animalId}</p>
+    </>
+  );
+}
+```
+
+## Complex Items
+Items within ListBox also allow for additional content used to better communicate options. Icons and descriptions can be added to the `children` of `Item` as shown in the example below. If a description is added, the prop `slot="description"` must be used to distinguish the different `<Text>` elements.
+
+```tsx example
+<ListBox width="size-2400" aria-label="Options" selectionMode="single">
+  <Section title="Permission">
+    <Item textValue="Read">
+      <Book size="S" />
+      <Text>Read</Text>
+      <Text slot="description">Read Only</Text>
+    </Item>
+    <Item textValue="Write">
+      <Draw size="S" />
+      <Text>Write</Text>
+      <Text slot="description">Read and Write Only</Text>
+    </Item>
+    <Item textValue="Admin">
+      <BulkEditUsers size="S" />
+      <Text>Admin</Text>
+      <Text slot="description">Full access</Text>
+    </Item>
+  </Section>
+</ListBox>
+```
+## Asynchronous loading
+
+ListBox supports loading data asynchronously, and will display a progress circle when the `isLoading` prop is set.
+It also supports infinite scrolling to load more data on demand as the user scrolls, via the `onLoadMore` prop.
+
+This example uses the [useAsyncList](../react-stately/useAsyncList.html) hook to handle loading the data.
+See the docs for more information.
+
+```tsx example
+import {useAsyncList} from '@react-stately/data';
+
+function AsyncLoadingExample() {
+  let list = useAsyncList({
+    async load({signal, cursor}) {
+      // If no cursor is available, then we're loading the first page.
+      // Otherwise, the cursor is the next URL to load, as returned from the previous page.
+      let res = await fetch(cursor || 'https://pokeapi.co/api/v2/pokemon', {signal});
+      let json = await res.json();
+      return {
+        items: json.results,
+        cursor: json.next
+      };
+    }
+  });
+
+  return (
+    <Flex maxHeight="size-2400">
+      <ListBox
+        label="Pick a Pokemon"
+        items={list.items}
+        isLoading={list.isLoading}
+        onLoadMore={list.loadMore}
+        width="size-2400">
+        {item => <Item key={item.name}>{item.name}</Item>}
+      </ListBox>
+    </Flex>
+  );
+}
+```
 
 ## Props
 
 <PropTable component={docs.exports.ListBox} links={docs.links} />
 
 ## Visual options
+### Loading
+
+```tsx example
+<ListBox
+  isLoading
+  aria-label="Choose frequency"
+  selectionMode="single"
+  onSelectionChange={selected => setFrequency(selected)}
+  width="size-1200">
+  <Item key="Rarely">Rarely</Item>
```

Comment:
This is kind of more of a loading more story right now rather than initial loading. Is that intended? Or should we remove these items?

Also, the middle option seems to get cut off a bit by the width.